cache/rediscache/rediscache.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package rediscache provides an implementation of prototransform.Cache
+// that is backed by a Redis instance: https://redis.io/.
+package rediscache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/bufbuild/prototransform"
+	"github.com/gomodule/redigo/redis"
```

Comment:
What do you think about packaging up the different implementations as different modules so consumers of prototransform only have to pull in what they need?

Reply:
The compiler already does this: consumer applications won't link in these packages if they don't use them. As far as trying to get them out of `go.mod` completely, (a) they only get downloaded if _needed_ (meaning they are imported by a package being built or their checksum is needed but not available in `go.sum` or checksum server), and (b) these two packages are _tiny_.

I think making lots of teeny modules just makes maintenance a pain and doesn't do anything in practice.

For example, when I `go get github.com/bufbuild/prototransform` into a new module, this is the output:
```
go get github.com/bufbuild/prototransform
go: downloading github.com/bufbuild/prototransform v0.0.0-20230512162240-aaf86165de13
go: downloading google.golang.org/protobuf v1.30.0
go: downloading buf.build/gen/go/bufbuild/reflect/bufbuild/connect-go v1.7.0-20230420145030-e29739ab7137.1
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/bufbuild/connect-go v1.7.0
go: downloading buf.build/gen/go/bufbuild/reflect/protocolbuffers/go v1.30.0-20230420145030-e29739ab7137.1
go: added buf.build/gen/go/bufbuild/reflect/bufbuild/connect-go v1.7.0-20230420145030-e29739ab7137.1
go: added buf.build/gen/go/bufbuild/reflect/protocolbuffers/go v1.30.0-20230420145030-e29739ab7137.1
go: added github.com/bufbuild/connect-go v1.7.0
go: added github.com/bufbuild/prototransform v0.0.0-20230512162240-aaf86165de13
go: added github.com/pkg/errors v0.9.1
go: added google.golang.org/protobuf v1.30.0
```
No references to redigo or memcache.

If I actually `go test .`, with a clean `$GOPATH/pkg`, again it does not download these modules, because they aren't actually used. Only if I do `go test ./...` (or build/test a program/package that actually imports the `cache/rediscache` or `cache/memcache` packages in here) will these dependencies be downloaded.